### ISSUE TITLE
Drives widget and sidebar not consistent when drives are added/removed

### ISSUE THREAD

**Issue by generalguy41** [files-community/Files#2245]
**Describe the bug**
Drives widget and sidebar don't update properly when drives are added/removed. Most commonly seen on the home page.

**To Reproduce**
1. Go to Home page.
2. Add or remove a drive.
3. Note the drives don't update.

**Expected behavior**
Drives should update.

**Screenshots**
1. Worked, but then crashed:
![ezgif-7-ba8575a12a05](https://user-images.githubusercontent.com/8879244/98322981-bd937a80-1fb6-11eb-808e-a31b0dfea51f.gif)
2. Didn't update, but didn't crash:
![ezgif-7-4458a1d8958a](https://user-images.githubusercontent.com/8879244/98323049-e6b40b00-1fb6-11eb-8cee-c36889cc5b2c.gif)
3. From a different directory, updated (plugging in), but then didn't update (unplugging). No crash:
![ezgif-7-9aab88d68ada](https://user-images.githubusercontent.com/8879244/98323109-1400b900-1fb7-11eb-9721-878b4896d27f.gif)

**Desktop**
 - OS Version: Windows 10 2004 20246.1
 - App version: v0.19.0.0

**Additional context**
- Clicking into the drive when removed obviously caused a crash.
- Also had straight crashes when plugging in or unplugging drives sometimes.

**Log file**
[debug.txt](https://github.com/files-community/Files/files/5498866/debug.txt)
[debug_fulltrust.txt](https://github.com/files-community/Files/files/5498867/debug_fulltrust.txt)

**Comment by generalguy41** [comment]
@gave92 This still occurs in 0.20.0.0. It doesn't seem to crash anymore, but unplugging a drive doesn't remove it from Files without a restart.

I tested again with 3 different USB drives, and it looks like the last one unplugged replaces the previous one.
![ezgif-7-de16948dfe7f](https://user-images.githubusercontent.com/8879244/98947883-bcd17b80-24c3-11eb-9143-f911f8563854.gif)

Also of note, when I plugged all three in at the same time, only the first ones size/progress bar updated.
![image](https://user-images.githubusercontent.com/8879244/98948218-2a7da780-24c4-11eb-843a-ec78d0f90762.png)

I can make a separate ticket for that, though it may be related to #2242.
